ADFA-2328 Expand the ribbon slider to a new position and make it as wide as the app

hal-eisen-adfa opened the pull request:
I had to change a lot more than I wanted to because it was hard to find what was injecting all the extra vertical whitespace. I'm open to suggestions on scaling back what I did.

I tested the long-press to tooltips on the ribbon slider toolbar - they still work.

Fixes #719

dara-abijo-adfa reviewed (commented):
I believe we may be able to achieve the desired outcome with fewer changes than are currently included in this PR. In particular, making updates to a small number of existing files (around five) without introducing new files may be sufficient.

I also noticed that the `.gitignore` file was modified as part of this PR, though it doesn’t appear to be directly related to the scope of this task. It might be worth revisiting whether that change is necessary here.

Lastly, the resulting UI may benefit from a bit of refinement, as the icons currently feel quite closely spaced.

Please let me know your thoughts, and I’m happy to discuss or clarify further.

jatezzz commented on app/src/main/java/com/itsaky/androidide/ui/CompactMaterialToolbar.kt:
```diff
@@ -0,0 +1,29 @@
+        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
+    }
+
+    override fun onLayout(changed: Boolean, l: Int, t: Int, r: Int, b: Int) {
```
this code is not needed

hal-eisen-adfa commented on common/src/main/java/com/itsaky/androidide/ui/ProjectActionsToolbar.kt:
```diff
@@ -57,21 +42,36 @@ class CustomToolbar @JvmOverloads constructor(
             setImageDrawable(icon)
             addCircleRipple()
+            // Remove any default padding from ImageButton
+            setPadding(0, 0, 0, 0)
```
Done, thanks

hal-eisen-adfa commented:
> I strongly feel we can achieve the result with less changes than we have in this PR. Specifically, with changes in about 5 of the existing files and no new files introduced, we should be good.
> 
> Also, I see the `.gitignore` file was modified in this PR, but it does not look related to this task.
> 
> Finally, the resulting UI could use some polishing; the icons are tightly spaced.

I've simplified the PR and removed one of the new files.

Can you show a screenshot of the icons being tightly spaced? They look ok on my S22 and my Pixel 9 emulator.
